### PR TITLE
Pupil not taking the check issues

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
- 18936 Fetch pupil attendance and display correct status if pupil is flagged as not taking the check
- 18960 Prevent pin generation for pupils that are not taking the check

### CODE REVIEW

**Comment by activemq** [comment]
looks good

File: admin/services/data-access/pupil-attendance.data.service.js [hunk 1]
```diff
@@ -70,5 +70,27 @@ pupilAttendanceDataService.findByPupilIds = async (ids) => {
   const sql = [select, whereClause].join(' ')
   return sqlService.query(sql, params)
 }
+/**
+ * Returns pupil attendance based on pupil id
+ * @param pupilId
+ * @return {Object}
+ */
+pupilAttendanceDataService.findByPupilId = async (pupilId) => {
```
Comment (jon-shipley) [inline admin/services/data-access/pupil-attendance.data.service.js]: Should be called `findOneById` please
This whole method can be replaced by a call to `sqlService.findOneById()`

File: admin/services/data-access/pupil-attendance.data.service.js [hunk 2]
```diff
@@ -70,5 +70,13 @@ pupilAttendanceDataService.findByPupilIds = async (ids) => {
   const sql = [select, whereClause].join(' ')
   return sqlService.query(sql, params)
 }
+/**
+ * Returns pupil attendance based on pupil id
+ * @param pupilId
+ * @return {Object}
+ */
+pupilAttendanceDataService.findOneByPupilId = async (pupilId) => {
+  return sqlService.findOneByPupilId(table, pupilId)
```
Comment (GuyHarwood) [inline admin/services/data-access/pupil-attendance.data.service.js]: ???  the sqlService is a generic data layer utility.  It should not have table specific methods

File: admin/services/data-access/sql.service.js [hunk 3]
```diff
@@ -282,6 +282,23 @@ sqlService.findOneById = async (table, id) => {
   return R.head(rows)
 }
 
+/**
+ * Find a row by numeric pupil ID
+ * @param {string} table
+ * @param {number} pupilId
+ * @return {Promise<void>}
+ */
+sqlService.findOneByPupilId = async (table, pupilId) => {
```
Comment (GuyHarwood) [inline admin/services/data-access/sql.service.js]: please remove this.  it belongs in a data service

File: admin/spec/service/pin.service.spec.js [hunk 4]
```diff
@@ -53,7 +53,9 @@ describe('pin.service', () => {
       // Because we used the pupil mock we expect the pupils to have the same name, so we need
       // show the dob to differentiate.
       expect(data[0].showDoB).toBe(true)
+      expect(data[0].dateOfBirth).toBe('9 Sep 2008')
       expect(data[1].showDoB).toBe(true)
+      expect(data[1].dateOfBirth).toBe('9 Sep 2008')
```
Comment (GuyHarwood) [inline admin/spec/service/pin.service.spec.js]: 👍

File: admin/tests-integration/sql.service.spec.js [hunk 5]
```diff
@@ -178,6 +178,17 @@ describe('sql.service:integration', () => {
     expect(typeof row).toBe('object')
   })
 
+  xit('#findOneByPupilId should retrieve a row', async () => {
```
Comment (GuyHarwood) [inline admin/tests-integration/sql.service.spec.js]: please remove. these should not be in the sql service

File: admin/services/data-access/pupil-attendance.data.service.js (hunk 1 above)
Comment (GeorgeChatzigiannis) [reply admin/services/data-access/pupil-attendance.data.service.js]: @jon-shipley updated

File: admin/services/data-access/pupil-attendance.data.service.js (hunk 2 above)
Comment (GeorgeChatzigiannis) [reply admin/services/data-access/pupil-attendance.data.service.js]: @GuyHarwood was wrong in the first place, updated

File: admin/services/data-access/sql.service.js (hunk 3 above)
Comment (GeorgeChatzigiannis) [reply admin/services/data-access/sql.service.js]: @GuyHarwood removed

File: admin/tests-integration/sql.service.spec.js (hunk 5 above)
Comment (GeorgeChatzigiannis) [reply admin/tests-integration/sql.service.spec.js]: @GuyHarwood removed

**Pull request merged**